library/item/item_setting/src/main/java/org/quiltmc/qsl/itemsetting/api/CustomItemSetting.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.itemsetting.api;
+
+import java.util.function.Supplier;
+
+import net.minecraft.item.Item;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.quiltmc.qsl.itemsetting.impl.CustomItemSettingImpl;
+
+/**
+ * A type of setting that can be passed to item constructors.
```

Comment:
what does "can be passed to item constructors" mean? it's not clear from this class alone.